presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSetTimezone.java

```diff
@@ -0,0 +1,384 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.jdbc;
+
+import com.google.common.io.Closer;
+import io.airlift.log.Logging;
+import io.prestosql.server.testing.TestingPrestoServer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+
+import static io.prestosql.jdbc.TestPrestoDriver.closeQuietly;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Checks that JDBC dates and timestamps are interpreted in the client JVM timezone and should not
+ * be affected by the session timezone.  We also compare the behavior to a reference JDBC driver.
+ */
+@Test(singleThreaded = true)
+public class TestJdbcResultSetTimezone
+{
+    private static final String OTHER_TIMEZONE = "Asia/Kathmandu";
+
+    private TestingPrestoServer server;
+    private List<ReferenceDriver> referenceDrivers;
+
+    private Connection connection;
+    private Statement statement;
+
+    @BeforeClass
+    public void setupServer()
+    {
+        assertNotEquals(OTHER_TIMEZONE, TimeZone.getDefault().getID(), "We need a timezone different from the default JVM one");
+        Logging.initialize();
+        server = TestingPrestoServer.create();
+        referenceDrivers = ImmutableList.of(new PostgresqlReferenceDriver()/*, new OracleReferenceDriver()*/);
```

Comment:
?

Reply:
BTW when i uncommented this, i got a failure
```
java.lang.AssertionError: ud.udsoac.oacdty: 178
	at oracle.jdbc.driver.T4CTTIdcb.fillupAccessors(T4CTTIdcb.java:1049)
	at oracle.jdbc.driver.T4CTTIdcb.receiveCommon(T4CTTIdcb.java:234)
	at oracle.jdbc.driver.T4CTTIdcb.receive(T4CTTIdcb.java:168)
	at oracle.jdbc.driver.T4C8Oall.readDCB(T4C8Oall.java:1040)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:482)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)
	at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:229)
	at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:41)
	at oracle.jdbc.driver.T4CStatement.executeForDescribe(T4CStatement.java:765)
	at oracle.jdbc.driver.OracleStatement.executeMaybeDescribe(OracleStatement.java:983)
	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1168)
	at oracle.jdbc.driver.OracleStatement.executeQuery(OracleStatement.java:1362)
	at oracle.jdbc.driver.OracleStatementWrapper.executeQuery(OracleStatementWrapper.java:369)
	at io.prestosql.jdbc.TestJdbcResultSetTimezone$OracleReferenceDriver.query(TestJdbcResultSetTimezone.java:293)
	at io.prestosql.jdbc.TestJdbcResultSetTimezone.checkRepresentation(TestJdbcResultSetTimezone.java:238)
	at io.prestosql.jdbc.TestJdbcResultSetTimezone.checkRepresentation(TestJdbcResultSetTimezone.java:231)
	at io.prestosql.jdbc.TestJdbcResultSetTimezone.testTime(TestJdbcResultSetTimezone.java:156)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
```

for testTime for UTC and Warsaw time zones

Reply:
i see value in keeping Oracle as the reference. Would it be possible to keep it except for TIME?